### ISSUE TITLE
False positive `unnecessary-comprehension` when modifying a dict to a list of tuples

### ISSUE THREAD

**Issue by martimlobao** [PyCQA/pylint#4499]
### Steps to reproduce

Given a file `a.py`:

```python
def main():
    mydict = {"foo": 1, "bar": 2}
    return [(k, v) for k, v in mydict.items()]
```

Pylint will complain about an unnecessary comprehension, even though the comprehension is modifying the type from a dictionary to a list of tuples.

### Current behavior

Result of `pylint a.py`:
```
$ pylint a.py 
************* Module example
example.py:3:0: R1721: Unnecessary use of a comprehension (unnecessary-comprehension)
```

### Expected behavior

No errors.

### pylint --version output

Result of `pylint --version` output:

```
$ pylint --version output
pylint 2.8.2
astroid 2.5.6
Python 3.8.7 (default, Jan 22 2021, 22:54:27) 
[Clang 12.0.0 (clang-1200.0.32.28)]
```

**EDIT:** I guess alternatively it might be cleaner to use `list(mydict.items())` instead. That feels like it might be a separate issue altogether, and I'm not sure if there aren't other false positives that can't be solved similarly. It's not that the comprehension is unnecessary, it's that there's a nicer way to do it. Maybe a new warning might also be a good idea?

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue !

**Comment by cdce8p** [comment]
I think this is more a case of a misleading error message that could be improved. As you pointed out `list(mydict.items())` is indeed equivalent and IMO the better solution. So technically this is an `unnecessary-comprehension`.

I can't remember the exact code, but I had something similar happen a while back where it took me more than a short moment to figure out a solution.

**Comment by yushao2** [comment]
Proposing an improvement to helptext for this checker in my PR, please take a look :)

**Comment by yushao2** [comment]
@Pierre-Sassoulas is it possible to assign this to me since I've already started some work on it? 

Thanks!
